src/components/Locations.js

```diff
@@ -0,0 +1,98 @@
+import { Component } from "react";
+import "../App.css";
+
+class Locations extends Component{
+    constructor(){
+        super();
+
+        this.state={
+            locations:[],
+            display:true,
+            locationName:"",
+            locationClimate:"",
+            locationTerrain:"",
+            buttonText: "Show Locations"
+        }
+    }
+  
+    fetchLocations=()=>{
+        fetch("https://ghibliapi.herokuapp.com/locations")
+            .then(res=>res.json())
+            .then ((data)=>{
+                this.setState({
+                    locations: data,
+            })
+        })
+    }
+
+    componentDidMount(){
+        this.fetchLocations();
+    }
+
+handleClick=(e)=>{
+    e.preventDefault();
+    let name = this.state.locations.map((location)=>location.name).join(", ");
+    let climate = this.state.locations.map((location)=>location.climate).join(", ");
+    let terrain = this.state.locations.map((location)=>location.terrain).join(", ");
+    this.setState({
+        locationName: name,
+        locationClimate: climate,
+        locationTerrain: terrain,
+        buttonText: "Hide Locations",
+
+    })
+}
+
+
+  render(){
+    return (
+      <div className="locations">
+          <h1>List of Locations</h1>
+            <button onClick={this.handleClick}>{this.state.buttonText}</button>
+            <ul>
+                <li>
+                     <strong>Name:</strong> {this.state.locationName}
+                </li>
+                <li>
+                    <strong>Climate:</strong>{this.state.locationClimate}
+                </li>
+                <li>
+                    <strong>Terrain:</strong>{this.state.locationTerrain}
+                </li>  
+            </ul>
+      </div> 
+    );
+  }
+}
```

Comment:
```suggestion
  render(){
  
  let locationsToDisplay = this.state.locations.map((location) => {
      return (
         <li>
             <div><strong>Name:</strong> {this.state.locationName}</div>
             <div><strong>Climate:</strong>{this.state.locationClimate}</div>
             <div><strong>Terrain:</strong>{this.state.locationTerrain}</div>
         </li>
     )
  })
  
    return (
      <div className="locations">
          <h1>List of Locations</h1>
            <button onClick={this.handleClick}>{this.state.display ? "Hide Locations" : "Show Locations"}</button>
            <ul>
               {this.state.display && locationsToDisplay}
            </ul>
      </div> 
    );
  }
}
```
1) You need to use `.map` to return an `li` for each location that includes its Name, Climate, and Terrain. 
2) The button's text content is entirely dependent on whether `this.state.display` is **true** or **false**. 
3) `{this.state.display && locationsToDisplay}` makes it so that the array of location `li`s will **only render if** `this.state.display` **is true**.

You were on the right track with your commented out code below!